src/hooks/useAxiosWithAuth0.js

```diff
@@ -2,50 +2,29 @@ import { useEffect } from 'react';
 import axios from 'axios';
 import { useAuth0 } from '@auth0/auth0-react';
 
-/*
-NOTE:
-React requires that we can only call hooks in function components or within other hooks. As axiosWithAuth is simply a helper function we've built a useAxiosWithAuth0() hook in which we've called both axiosWithAuth AND useAuth0().
-axiosWithAuth serves the same function as it typically does. useAuth0() gives access to getAccessTokenSilently() for authentication/authorization purposes. 
-This hook below is exported and used throughout the app in place of axiosWithAuth.
-THIS useAxiosWithAuth0 HOOK SIMPLY SERVES TO HELP CONNECT THE AXIOS CALL AND THE AUTHENTICATION.
-*/
+const axios_instance = axios.create({
+  baseURL: process.env.REACT_APP_API_URI,
+});
 
-export default function useAxiosWithAuth0() {
-  const { getAccessTokenSilently } = useAuth0();
-
-  const axiosWithAuth = token => {
-    return axios.create({
-      baseURL: process.env.REACT_APP_API_URI,
-      headers: {
-        authorization: `Bearer ${token}`,
-      },
-    });
-  };
+const useAxiosWithAuth0 = () => {
+  const { getAccessTokenSilently, isAuthenticated } = useAuth0();
 
   useEffect(() => {
-    const locations = [
-      '/',
-      '/apply',
-      '/apply/mentee',
-      '/apply/mentor',
-      '/apply/success',
-    ];
-
-    /********************************** */
+    if (isAuthenticated) {
+      axios_instance.interceptors.request.use(
```

Comment:
can you cross link to the axios docs in a code comment somewhere here in the file? This is brilliant! https://axios-http.com/docs/interceptors

After talking through this logic surrounding `isAuthenticated` and learning a bit about axios.interceptors I feel confident this is the closest to the final solution we'd like too see!